src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -245,8 +245,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ItemGroup>
 
       <!-- Copy all the assemblies -->
-      <ResolvedFileToPublish Include="@(ResolvedAssembliesToPublish)">
-      </ResolvedFileToPublish>
+        <ResolvedFileToPublish Include="@(ResolvedAssembliesToPublish)" Condition="'$(SkipRemovingSystemFiles)' == 'true' Or !$([System.Text.RegularExpressions.Regex]::IsMatch(%(Filename), '^System.'))" />
```

Comment:
great catch, but instance methods can not be used inline in msbuild, we need a static method.
I will escape the '.' for now